src/packet_analysis/Analyzer.cc

```diff
@@ -68,34 +68,60 @@ bool Analyzer::IsAnalyzer(const char* name)
 	return packet_mgr->GetComponentName(tag) == name;
 	}
 
+const AnalyzerPtr& Analyzer::LookupAnalyzer(uint32_t identifier) const
+	{
+	return dispatcher.LookupAnalyzer(identifier);
+	}
+
 AnalyzerPtr Analyzer::Lookup(uint32_t identifier) const
 	{
-	return dispatcher.Lookup(identifier);
+	return LookupAnalyzer(identifier);
 	}
 
-bool Analyzer::ForwardPacket(size_t len, const uint8_t* data, Packet* packet,
-                             uint32_t identifier) const
+const AnalyzerPtr& Analyzer::LookupAnalyzer(size_t len, const uint8_t* data, Packet* packet,
+                                            uint32_t identifier) const
+	{
+	const auto& identifier_based_analyzer = LookupAnalyzer(identifier);
+	if ( identifier_based_analyzer )
+		return identifier_based_analyzer;
+
+	const auto& detect_based_analyzer = DetectAnalyzer(len, data, packet);
+	if ( detect_based_analyzer )
+		return detect_based_analyzer;
+
+	return default_analyzer;
+	}
+
+const AnalyzerPtr& Analyzer::LookupAnalyzer(size_t len, const uint8_t* data, Packet* packet) const
 	{
-	auto inner_analyzer = Lookup(identifier);
-	if ( ! inner_analyzer )
+	const auto& detect_based_analyzer = DetectAnalyzer(len, data, packet);
+	if ( detect_based_analyzer )
+		return detect_based_analyzer;
+
+	return default_analyzer;
+	}
+
+const AnalyzerPtr& Analyzer::DetectAnalyzer(size_t len, const uint8_t* data, Packet* packet) const
+	{
+	for ( const auto& child : analyzers_to_detect )
 		{
-		for ( const auto& child : analyzers_to_detect )
+		if ( child->DetectProtocol(len, data, packet) )
 			{
-			if ( child->DetectProtocol(len, data, packet) )
-				{
-				DBG_LOG(DBG_PACKET_ANALYSIS,
-				        "Protocol detection in %s succeeded, next layer analyzer is %s",
-				        GetAnalyzerName(), child->GetAnalyzerName());
-				inner_analyzer = child;
-				break;
-				}
+			DBG_LOG(DBG_PACKET_ANALYSIS,
+			        "Protocol detection in %s succeeded, next layer analyzer is %s",
+			        GetAnalyzerName(), child->GetAnalyzerName());
+			return child;
 			}
 		}
+	return nil;
+	}
 
-	if ( ! inner_analyzer )
-		inner_analyzer = default_analyzer;
+bool Analyzer::ForwardPacket(size_t len, const uint8_t* data, Packet* packet,
+                             uint32_t identifier) const
+	{
+	const auto& analyzer = LookupAnalyzer(len, data, packet, identifier);
```

Comment:
The other naming nit: Staying with `inner_analyzer` or something like `next_analyzer` would be more descriptive here.